mammon/ext/ircv3/sasl.py

```diff
@@ -40,23 +40,45 @@ def m_AUTHENTICATE(cli, ev_msg):
     if len(ev_msg['params']) == 1 and ev_msg['params'][0] == '*':
         if getattr(cli, 'sasl', None):
             cli.dump_numeric('906', ['SASL authentication aborted'])
-            cli.sasl = None
         else:
             cli.dump_numeric('904', ['SASL authentication failed'])
+        cli.sasl = None
+        cli.sasl_tmp = ''
         return
 
     if getattr(cli, 'sasl', None):
-        if len(ev_msg['params'][0]) > 400:
+        raw_data = ev_msg['params'][0]
+
+        if len(raw_data) > 400:
             cli.dump_numeric('905', ['SASL message too long'])
             cli.sasl = None
+            cli.sasl_tmp = ''
+            return
+        elif len(raw_data) == 400:
+            if not hasattr(cli, 'sasl_tmp'):
+                cli.sasl_tmp = ''
+            cli.sasl_tmp += raw_data
+            # allow 4 'continuation' lines before rejecting for length
+            if len(cli.sasl_tmp) > 400 * 4:
```

Comment:
Could if we really wanted to, but I don't think users can REG a passphrase
long enough to overrun this right now. (Though, given we accept 2048 rather
than 512... might do it actually. Don't think I took that into account)

Once we have other mechanisms than PLAIN, might be good to allow larger
values too.

On Saturday, January 30, 2016, Valentin Lorentz <notifications@github.com>
wrote:

> In mammon/ext/ircv3/sasl.py
> <https://github.com/mammon-ircd/mammon/pull/89#discussion_r51342828>:
>
> >              cli.dump_numeric('905', ['SASL message too long'])
> >              cli.sasl = None
> > +            cli.sasl_tmp = ''
> > +            return
> > +        elif len(raw_data) == 400:
> > +            if not hasattr(cli, 'sasl_tmp'):
> > +                cli.sasl_tmp = ''
> > +            cli.sasl_tmp += raw_data
> > +            # allow 4 'continuation' lines before rejecting for length
> > +            if len(cli.sasl_tmp) > 400 * 4:
>
> Could you make this configurable?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mammon-ircd/mammon/pull/89/files#r51342828>.
>